xio-core/src/test/java/com/xjeffrose/xio/http/EdgeProxyFunctionalTest.java

```diff
@@ -103,29 +89,48 @@ public RouteStates(Function<U, T> factory, RouteConfigs configs) {
     public EdgeProxyConfig(Config config) {
       super(config);
       List<Config> routes = (List<Config>) config.getConfigList("routes");
+
       routeConfigs =
-          new RouteConfigs<ProxyRouteConfig>((Config c) -> new ProxyRouteConfig(c), routes);
+          new RouteConfigs<>(
+              routes.stream().map(ProxyRouteConfig::new).collect(Collectors.toList()));
       allPermissions = null;
     }
   }
 
   public class EdgeProxyState extends ApplicationState {
 
-    private final RouteStates routeStates;
+    private final RouteStates<ProxyRouteState> routeStates;
     private final ProxyClientFactory clientFactory;
 
+    public <T, K, U> Collector<T, ?, Map<K, U>> toLinkedMap(
+        Function<? super T, ? extends K> keyMapper, Function<? super T, ? extends U> valueMapper) {
+
+      return Collectors.toMap(
+          keyMapper,
+          valueMapper,
+          (u, v) -> {
```

Comment:
What is the 'v' in this context? Could we add comments on what this is? Does the 'u' here map to the generic 'U' in the method declaration?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-java.util.function.BinaryOperator-java.util.function.Supplier-

This is the merge function that gets called by `toMap` when the stream produces two entries with the same key. For us this is an error condition. But there's a conversation to be had about how this needs to go away in favor of a "route map" that doesn't care about the ordering of the keys.

To your previous point, `v` could be `ignored`.

Reply:
Ok, great. Let's do 'ignored' then.